stdlib/map.mli

```diff
@@ -86,6 +86,16 @@ module type S =
        of [x] in [m] disappears.
        @before 4.03 Physical equality was not ensured. *)
 
+    val update: key -> ('a option -> 'a) -> 'a t -> 'a t
```

Comment:
It would good to ensure that if we return the same option (physically, or perhaps checking physical equality under the `Some`), the function returns the original map (physically) and doesn't allocate anything.